browser/browser_test.go

```diff
@@ -0,0 +1,273 @@
+	FailNotNil(err, t)
+	defer os.RemoveAll(temp)
+	// Make two directories in temp
+	err = os.Mkdir(filepath.Join(temp, "a"), 0777)
+	FailNotNil(err, t)
+	err = os.Mkdir(filepath.Join(temp, "b"), 0777)
+	FailNotNil(err, t)
+	go List("test", outChan, temp)
+	res := (<-outChan)
+	if res.Err != "" {
+		t.Fail()
+	}
+	if len(res.Files) != 2 {
+		t.Fail()
+	}
+}
+
+func TestListNotExist(t *testing.T) {
+	var res *ResultSet
+	go List("test", outChan, "does/not/exist")
+	res = (<-outChan)
+	if res.Err == "" {
+		t.Log("Error should occur.")
+		t.Fail()
+	}
+}
+
+func TestMakeDirectoryAndRemove(t *testing.T) {
+	home, err := ioutil.TempDir("", "MakeAndRemove")
+	FailNotNil(err, t)
+	paths := []string{"/test_folder", "/test_folder/sub_folder"}
+	var res *ResultSet
+
+	defer func() {
+		Remove("test", dummyChan, filepath.Join(home, paths[0]))
+	}()
+
+	for _, p := range paths {
+		go MakeDirectory("test", outChan, filepath.Join(home, p))
+		res = (<-outChan)
+		if res.Err != "" {
+			t.Fatal(res.Err)
+		}
+		if !Exists(filepath.Join(home, p)) {
+			t.Fatal("Directory not created")
+			t.Fail()
+		}
+	}
+}
+
+func TestMakeDirectoryBadPath(t *testing.T) {
+	go MakeDirectory("test", outChan, "does/not/exist")
+	res := <-outChan
+	if res.Err == "" {
+		t.Fatal("Error should occur.")
+	}
+}
+
+func TestCopy(t *testing.T) {
+	home, err := ioutil.TempDir("", "copy")
+	FailNotNil(err, t)
+
+	dir := []string{"copy_folder", "copy_folder/sub1",
+		"copy_folder/sub2", "copy_folder/sub1/sub3",
+		"copy_to"}
+
+	// Create a directory for copying
+	for _, p := range dir {
+		MakeDirectory("test", dummyChan, filepath.Join(home, p))
+	}
+
+	d1 := filepath.Join(home, "copy_folder")
+	d2 := filepath.Join(home, "copy_to/")
+
+	defer func() {
+		_ = os.RemoveAll(d1)
+		_ = os.RemoveAll(d2)
+	}()
+
+	Copy("test", dummyChan, d1, d2)
+	WaitForOperationsToComplete()
+
+	if CompareDirectory(d1, filepath.Join(d2, "copy_folder")) == false {
+		t.Fatal("Directories not similar.")
+	}
+}
+
+func TestCopySrcNotExist(t *testing.T) {
+	go Copy("test", outChan, "does/not/exist", os.TempDir())
+	res := (<-outChan)
+	if res.Err == "" {
+		t.Fatal(res.Err)
+	}
+}
+
+func TestCopyDestNotExist(t *testing.T) {
+	f, err := ioutil.TempFile("", "existent_source")
+	FailNotNil(err, t)
+	f.Close()
+	go Copy("test", outChan, f.Name(), "does/not/exist")
+	res := (<-outChan)
+	if res.Err == "" {
+		t.Fatal(res.Err)
+	}
+}
+
+func TestGetFile(t *testing.T) {
+
+	temp, err := ioutil.TempDir("", "GetFile")
+	FailNotNil(err, t)
+	tf, err := ioutil.TempFile(temp, "getfile")
+	FailNotNil(err, t)
+
+	// File paths
+	fp := tf.Name()
+
+	defer func() {
+		_ = os.RemoveAll(temp)
+		if tf != nil {
+			_ = tf.Close()
+		}
+		_ = os.RemoveAll(fp)
+	}()
+
+	data := []byte{}
+	compBuff := []byte{}
+
+	gen := rand.New(rand.NewSource(1))
+	size := 3000
+	for i := 0; i < int(size); i++ {
+		num := gen.Int31()
+		for j := 0; j < 4; j++ {
+			data = append(data, byte(num<<uint(j)*8))
```

Comment:
@walac: Would there be any advantage of using big-endian by changing 8 to 0xff000000 ?